corehq/apps/reports_core/filters.py

```diff
@@ -261,13 +264,33 @@ def __init__(self, name, field, datatype, label, show_all, url_generator, choice
         self.url_generator = url_generator
         self.choice_provider = choice_provider
 
+    def context(self, values, lang=None):
+        context = super(DynamicChoiceListFilter, self).context(values, lang)
+        context['value'] = self._values_to_dict(values)
+        return context
+
+    def _values_to_dict(self, values):
+        """Some values are returned as Choice objects which need to be converted to
+        dicts to be displayed properly
```

Comment:
The only reason this wasn't an issue before, is because DynamicChoiceProviders never had defaults set.